### PR TITLE
Sp/rhub api

### PR DESCRIPTION
**Pull request by spodgorny9**
Add code to access research hub API

### CODE REVIEW

File: elm/web/rhub.py [hunk 1]
```diff
@@ -4,626 +4,788 @@
...
-        n_pages : int
-            Number of pages to get from the API. Typical response has 50
-            entries per page. Default of 1 ensures that this class doesnt hang
-            on a million responses.
-        txt_dir : str
-            File directory where you would like to save output .txt files.
+        record : dict
+            Profile in dict form, typically a response from the API.
         """
+        api_key = os.getenv("RHUB_API_KEY")
```
Comment (grantbuster) [inline elm/web/rhub.py:25]: In the class init docstrings for this class (and the other classes that use this) can you specify you need to set RHUB_API_KEY as an env variable and write a note on where to get this?

File: elm/web/rhub.py [hunk 2]
```diff
@@ -4,626 +4,788 @@
...
+        records = self._get_first()
+        for page in self._get_pages(n_pages=n_pages):
+            records += page
+        records = [ProfilesRecord(single) for single in records]
+        super().__init__(records)
 
-    def build_meta(self):
-        """Build a meta dataframe containing relevant information for
-        researchers.
+    def _get_first(self):
```
Comment (grantbuster) [inline elm/web/rhub.py:328]: Recommend wrapping _get_first and _get_pages in a single function so it's easy to _get_all()

You could do this in the osti module too (i realize you're just copying my mediocre implementation).

**Review by grantbuster — CHANGES_REQUESTED** [review]
Minor changes

File: elm/web/rhub.py (hunk 1 above)
Comment (spodgorny9) [reply elm/web/rhub.py:29]: I reached out to Rob about public access to the API and I'm still waiting to hear back so I just put in a placeholder for now.

File: examples/research_hub/retrieve_docs.py [hunk 3]
```diff
@@ -51,6 +51,37 @@
...
+    if pdf_obj.is_double_col():
+        text = pdf_obj.clean_poppler(layout=False)
+    text = pdf_obj.clean_headers(char_thresh=0.6, page_thresh=0.8,
+                                 split_on='\n',
+                                 iheaders=[0, 1, 3, -3, -2, -1])
+    with open(txt_fp, 'w') as f:
+        f.write(text)
+    logger.info(f'Saved: {txt_fp}')
+
+    return text
```
Comment (grantbuster) [inline examples/research_hub/retrieve_docs.py:82]: Can you make this a class method of PDFtoTXT?

**Review by grantbuster — CHANGES_REQUESTED** [review]

File: elm/web/rhub.py (hunk 1 above)
Comment (grantbuster) [reply elm/web/rhub.py:29]: Can you just say "contact an NREL library representative" if this is how we get the key?

**Review by grantbuster — APPROVED** [review]

**Pull request merged**